src/ast/variables/ExportDefaultVariable.ts

```diff
@@ -76,7 +92,14 @@ export default class ExportDefaultVariable extends LocalVariable {
 			: original.getName(getPropertyAccess);
 	}
 
-	getOriginalVariable(): Variable {
+	private previousOriginalVariableState: OriginalVariableStage | null = null;
+	getOriginalVariable(
+		originalVariableState: OriginalVariableStage = OriginalVariableStage.GENERATE_CHUNK
+	): Variable {
+		if (this.previousOriginalVariableState !== originalVariableState) {
```

Comment:
https://github.com/rollup/rollup/pull/5443/files#diff-9eb57702aec40721397a4dce9941382b2041e026149026ffe675b301128f2098R11-R12

```typescript
// Between two stages, there might be variable bound change, like `export default a; a = 0`.
// Therefore we clear the cache if the stage changes.
```

We might get a 'wrong'(not complete but sound for function call tracking) original variable in bind reference stage, so just clear the cache if it's in the new stage.

Reply:
Interesting. I would have hoped that such scenarios always trigger another tree-shaking pass so that there is always one last pass with the "correct" values.